src/user.ts

```diff
@@ -0,0 +1,232 @@
+            return undefined
+        }
+
+        const urlSafeOrgName = orgName.toLowerCase().replace(/ /g, "-")
+        for (const orgId in this.orgIdToUserOrgInfo) {
+            const orgMemberInfo = this.orgIdToUserOrgInfo[orgId]
+            if (orgMemberInfo?.urlSafeOrgName === urlSafeOrgName) {
+                return orgMemberInfo
+            }
+        }
+
+        return undefined
+    }
+
+    public getUserProperty(key: string): unknown | undefined {
+        if (!this.properties) {
+            return undefined
+        }
+
+        return this.properties[key]
+    }
+
+    public getOrgs(): UserOrgInfo[] {
+        if (!this.orgIdToUserOrgInfo) {
+            return []
+        }
+
+        return Object.values(this.orgIdToUserOrgInfo)
+    }
+
+    public isImpersonating(): boolean {
+        return !!this.impersonatorUserId
+    }
+
+    public isRole(orgId: string, role: string): boolean {
+        const orgMemberInfo = this.getOrg(orgId)
+        if (!orgMemberInfo) {
+            return false
+        }
+
+        return orgMemberInfo.isRole(role)
+    }
+
+    public isAtLeastRole(orgId: string, role: string): boolean {
+        const orgMemberInfo = this.getOrg(orgId)
+        if (!orgMemberInfo) {
+            return false
+        }
+
+        return orgMemberInfo.isAtLeastRole(role)
+    }
+
+    public hasPermission(orgId: string, permission: string): boolean {
+        const orgMemberInfo = this.getOrg(orgId)
+        if (!orgMemberInfo) {
+            return false
+        }
+
+        return orgMemberInfo.hasPermission(permission)
+    }
+
+    public hasAllPermissions(orgId: string, permissions: string[]): boolean {
+        const orgMemberInfo = this.getOrg(orgId)
+        if (!orgMemberInfo) {
+            return false
+        }
+
+        return orgMemberInfo.hasAllPermissions(permissions)
+    }
+
+    public static fromJSON(json: string): User {
+        const obj = JSON.parse(json)
+        const orgIdToUserOrgInfo: OrgIdToUserOrgInfo = {}
+        for (const orgId in obj.orgIdToUserOrgInfo) {
+            orgIdToUserOrgInfo[orgId] = UserOrgInfo.fromJSON(JSON.stringify(obj.orgIdToUserOrgInfo[orgId]))
+        }
+        try {
+            return new User(
+                obj.userId,
+                obj.email,
+                orgIdToUserOrgInfo,
+                obj.firstName,
+                obj.lastName,
+                obj.username,
+                obj.legacyUserId,
+                obj.impersonatorUserId,
+                obj.properties
+            )
+        } catch (e) {
+            console.error("Unable to parse User. Make sure the JSON string is a stringified `User` type.", e)
+            throw e
+        }
+    }
+}
+
+interface OrgIdToUserOrgInfo {
+    [orgId: string]: UserOrgInfo
+}
+
+export class UserOrgInfo {
+    public orgId: string
+    public orgName: string
+    public orgMetadata: { [key: string]: any }
+    public urlSafeOrgName: string
+
+    private userAssignedRole: string
```

Comment:
We ran into some awkward type issues when making these fields private before. I don't think there's any reason they need to be private (and it's not super enforced anyways) so maybe we just leave these public?